App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/Employer/ManageJob/JobSummaryCard.jsx

```diff
@@ -24,25 +24,81 @@ export class JobSummaryCard extends React.Component {
-            let { city, country } = location;
-            if (city && country) {
-                locationMeta = `${city}, ${country}`;
-            } else if (city) {
-                locationMeta = city;
-            } else if (country) {
-                locationMeta = country;
-            }
-        }
+        if (!location) return '';
 
+        const { city, country } = location; 
+        
+        const locationMeta = city && country ? `${city}, ${country}` : city || country || '';
+        
         return locationMeta;
     }
 
-    init() {
-        this.props.jobData.forEach(job => {
-            let card = (
-                <Card key={job.id} fluid >
+    componentDidMount() {
+        // this.init();
+    };
+
+    selectJob(id) {
+        const cookies = Cookies.get('talentAuthToken');
+        //url: 'https://talentservicestalentanderson.azurewebsites.net/listing/listing/GetJobByToEdit',
+    }
+
+    displayCards() {
+        return this.state.cards;
+    }
+
+    closeJob(id) {
+        const url = 'https://talentservicestalentanderson.azurewebsites.net/listing/listing/closeJob';
+        const cookies = Cookies.get('talentAuthToken');
+
+        try {
+            if (!id) {
+                TalentUtil.notification.show("Job ID is missing", "error", null, null)
+                return;
+            }
+
+            $.ajax({
+                url: url,
+                headers: {
+                    'Authorization': 'Bearer ' + cookies,
+                    'Content-Type': 'application/json'
+                },
+                type: "POST",
+                contentType: "application/json",
+                data: JSON.stringify(id),
+                dataType: "json",
+                success: function (res) {
+                    try {
+                        if (res.success == true) {
+                            TalentUtil.notification.show(res.message, "success", null, null);
+                            window.location = "/ManageJobs";
+                           
+                        } else {
+                            TalentUtil.notification.show(res.message, "error", null, null)
+                        }
+                    } catch (err) {
+                        console.error(err);
+                        TalentUtil.notification.show("An error occured. Please try again", "error", null, null)
+                    }
+                    
+                }.bind(this),
+                error: function (res) {
+                    TalentUtil.notification.show(res.message, "error", null, null)
+                }
+            })
+
+        } catch (err) {
+            console.error(err);
+            TalentUtil.notification.show("An error occured. Please try again", "error", null, null)
+        }
+    }
+
+    render() {
+        if (!Array.isArray(this.props.jobData) || !this.props.jobData.length) {
+            return null;
+        }
+
+        const cards = this.props.jobData.map(job => (
```

Comment:
If you are handling data and going to render them, always make sure that you are treating the content correctly. Let's say for example **jobData** is an array, but, let's say the BE changes and it became an object, or accidentally return a null or empty string. If that happens, your application might display some errors. Before you map an array, always make sure that this array field is actually an array. You can check it by adding a condition like **this.props?.jobData?.map.length** and then continue with your code,